src/hotspot/share/runtime/threadSMR.inline.hpp

```diff
@@ -33,6 +33,59 @@
 #include "memory/iterator.hpp"
 #include "runtime/prefetch.inline.hpp"
 #include "runtime/thread.inline.hpp"
+#include "utilities/debug.hpp"
+#include "utilities/macros.hpp"
+
+ThreadsList::Iterator::Iterator(ThreadsList* list, uint i) :
+  _thread_ptr(list->threads() + check_index(list, i))
+  DEBUG_ONLY(COMMA _list(list))
+{}
+
+bool ThreadsList::Iterator::operator==(Iterator i) const {
+  assert_not_singular();
+  assert_same_list(i);
+  return _thread_ptr == i._thread_ptr;
+}
+
+bool ThreadsList::Iterator::operator!=(Iterator i) const {
+  assert_not_singular();
+  assert_same_list(i);
+  return _thread_ptr != i._thread_ptr;
```

Comment:
Better might be to have the entire body consist of `return operator==(i);`.

Reply:
I think what's there is more clear. If we switch to `return operator==(i)`,
then I think that will lead to some head scratching. Especially since it's
making me wonder right now...

Reply:
And _obviously_ I meant `return !operator==(i)`.  Need to keep that all-important one-character difference...

Reply:
Thanks for the clarification. Now you see why I was scratching my head and
wondering what C++ voodoo you were doing now... :-)